### PR TITLE
Adding Extension Signer to SDK 4.0

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### _Summary_

Added extension support for SCW

### CODE REVIEW

**Review by Jacekdaa — COMMENTED** [review]
ok

File: packages/wallet-sdk/src/sign/SignerConfigurator.ts [hunk 1]
```diff
@@ -76,6 +77,8 @@ export class SignerConfigurator {
       this.initScwSigner();
     } else if (this.signerType === 'walletlink') {
       this.initWalletLinkSigner();
+    } else if (this.signerType === 'extension') {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/SignerConfigurator.ts:80]: let's use `switch`

File: packages/wallet-sdk/src/sign/scw/transport/PopUpCommunicator.ts [hunk 2]
```diff
@@ -68,11 +68,18 @@ export class PopUpCommunicator extends CrossDomainCommunicator {
...
-      this.popUpConfigurator.resolveSignerTypeSelection = resolve;
+  selectSignerType({
+    smartWalletOnly,
+    isExtensionAvailable,
+  }: {
+    smartWalletOnly: boolean;
+    isExtensionAvailable: boolean;
+  }): Promise<SignerType> {
+    return new Promise((resolve, reject) => {
+      this.popUpConfigurator.signerTypeSelectionFulfillment = { resolve, reject };
```
Comment (fan-zhang-sv) [inline packages/wallet-sdk/src/sign/scw/transport/PopUpCommunicator.ts:79]: this is to unblock user for when they close the popup during onboarding

File: packages/wallet-sdk/src/sign/extension/ExtSigner.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      onAccountsUpdate: (...args) => options.updateListener.onAccountsUpdate(this, ...args),
+      onChainUpdate: (...args) => options.updateListener.onChainUpdate(this, ...args),
+    };
+    this.adapter = options.adapter;
+    this.adapter.setAppInfo?.(options.appName, options.appLogoUrl);
+  }
+
+  async handshake(): Promise<AddressString[]> {
+    const newAddresses = await this.request<AddressString[]>({ method: 'eth_requestAccounts' });
+    this.updateListener.onAccountsUpdate({
```
[the remark this answers is not in the document]
Comment (bangtoven) [reply packages/wallet-sdk/src/sign/extension/ExtSigner.ts:28]: what's alternative? @fan-zhang-sv

[the remark this answers is not in the document]
Comment (fan-zhang-sv) [reply packages/wallet-sdk/src/sign/extension/ExtSigner.ts:28]: tbh i don't think theres any, we need to use these updateListern cause we'd want to update the state inside the new Provider

File: packages/wallet-sdk/src/CoinbaseWalletSDK.ts [hunk 4]
```diff
@@ -44,34 +46,26 @@ export class CoinbaseWalletSDK {
   }
 
   public makeWeb3Provider(): ProviderInterface {
-    if (!this.smartWalletOnly) {
-      const extension = this.walletExtension;
-      if (extension) {
-        extension.setAppInfo?.(this.appName, this.appLogoUrl);
-        return extension;
-      }
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletSDK.ts:52]:
```
if (extension && !window.injectedSigner) {
```

**Pull request closed without merging**